plugins/HydrogenImport/LocalFileMng.cpp

```diff
@@ -197,10 +196,7 @@ QDomDocument LocalFileMng::openXmlDocument( const QString& filename )
 		return QDomDocument();
 
 	if( TinyXMLCompat ) {
-	    QString enc = QTextCodec::codecForLocale()->name();
-	    if( enc == QString("System") ) {
-		    enc = "UTF-8";
-	    }
+		const QString enc = "UTF-8"; // unknown encoding, so assume utf-8 and call it a day
```

Comment:
I'm not sure how common it is for this XML to be an encoding other than UTF8, so I don't know the severity of this change, however from what I'm reading, it's pretty bleak.  Newer versions of Qt6 will have ICU (International Components for Unicode) support, but this comment suggests that ICU support won't be landing on Windows anytime soon.  https://forum.qt.io/topic/138790/replacing-calls-to-qtextstream-setcodec-in-qt-6/11?_=1729692809826&lang=en-US

We may -- if we decide to, add a comment that references the upstream feature incase this UTF-8 assumption causes an unobvious code regression.

```cpp
// Qt 6.4+ (https://codereview.qt-project.org/c/qt/qtbase/+/393373)
QStringConverter(...) ...
```

Reply:
What should also be noted is that `QString enc = QTextCodec::codecForLocale()->name();` seems to make an odd assumption that the current encoding of the current locale is suitable for parsing an XML, which seems -- at least at a glance -- an odd and bug-prone assumption.  Hard-coding UTF-8 seems more predictable, even if it causes some edge-case regressions where the system local and the XML are both coincidentally in a non-UTF8 encoding.

Reply:
You are correct. As I mention in the commit message, the original choice to use the "best match" encoding from the locale is error prone because there is no guarantee that the encoding used to save the file will be the same encoding of the user importing it in LMMS.

There is no good solution here, so I decided to simply assume the old Hydrogen files are UTF-8 encoded. The worst that can happen is someone's ancient Hydrogen files containing non-ASCII text of some random encoding becomes mojibake'd after importing into LMMS, which is something that already could have happened.

Reply:
Thanks.  Yes, I see how that the original PR already uncovered this. Sorry for any redundancies that it caused.